spec/mongoid/clients/factory_spec.rb

```diff
@@ -389,16 +397,19 @@
 
     let(:config) do
       {
-        default: { hosts: [ "127.0.0.1:1234" ], database: database_id},
+        default: { hosts: SpecConfig.instance.addresses, database: database_id },
         good_one: { hosts: [ "127.0.0.1:1234" ], database: database_id},
         bad_one: { hosts: [ "127.0.0.1:1234" ], database: database_id}.merge(unknown_opts),
-        good_two: { uri: "mongodb://127.0.0.1:1234,127.0.0.1:5678/#{database_id}?serverSelectionTimeoutMS=1000" },
-        bad_two: { uri: "mongodb://127.0.0.1:1234,127.0.0.1:5678/#{database_id}?serverSelectionTimeoutMS=1000" }.merge(unknown_opts)
+        good_two: { uri: "mongodb://127.0.0.1:1234,127.0.0.1:5678/#{database_id}" },
+        bad_two: { uri: "mongodb://127.0.0.1:1234,127.0.0.1:5678/#{database_id}" }.merge(unknown_opts)
       }
     end
 
-    before do
+    around(:each) do |example|
+      old_config = Mongoid::Config.clients
       Mongoid::Config.send(:clients=, config)
+      example.run
+      Mongoid::Config.send(:clients=, old_config)
```

Comment:
I think this line needs to be in an `ensure` block otherwise it won't run if the test fails.

Reply:
This is how I did it initially, but it turned out that `around` hook always executes. Even in case of an exception raised inside an example. It is mentioned in rspec docs - https://relishapp.com/rspec/rspec-core/v/3-8/docs/hooks/around-hooks#an-around-hook-continues-to-run-even-if-the-example-throws-an-exception

We still can add an `ensure` block here, but it might be confusing.

Reply:
I see, this is correct then.